### PR TITLE
`use_decorated_box`

### PR DESCRIPTION
**Pull request by minhqdao**
## Description

Use `DecoratedBox` over `Container` when only `Decoration` is specified.

Fixes https://github.com/dart-lang/linter/issues/3060

## Kind

Style advice.

## Bad Example

```dart
Widget buildArea() {
  return Container(
    decoration: const BoxDecoration(
      color: Colors.blue,
      borderRadius: BorderRadius.all(
        Radius.circular(5),
      ),
    ),
    child: const Text('...'),
  );
}
```

## Good Example

```dart
Widget buildArea() {
  return const DecoratedBox(
    decoration: BoxDecoration(
      color: Colors.blue,
      borderRadius: BorderRadius.all(
        Radius.circular(5),
      ),
    ),
    child: Text('...'),
  );
}
```

### CODE REVIEW

File: test_data/rules/use_decorated_box.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N use_decorated_box`
```
Comment (bwilkerson) [inline test_data/rules/use_decorated_box.dart:5]: We should also have a test case in which a `Container` is created with a `decoration` and at least one other argument other than `key` or `child`.

**Review by bwilkerson — COMMENTED** [review]
Other than the one missing test case, this looks good.

**Review by bwilkerson — APPROVED** [review]

File: lib/src/rules/use_decorated_box.dart [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../util/flutter_utils.dart';
+
+const _desc = r'Use `DecoratedBox`.';
+
+const _details = r'''
+
+**DO** use `DecoratedBox` when `Container` has only `Decoration`.
```
Comment (pq) [inline lib/src/rules/use_decorated_box.dart:15]: Consider:

    **DO** use `DecoratedBox` when `Container` has only a `Decoration`.

**Review by pq — CHANGES_REQUESTED** [review]
Looks great. And thanks for kicking off the thoughtful corresponding conversation in #3060! 👍 

Could you add an attribution for yourself in `AUTHORS`?

Thanks!

File: lib/src/rules/use_decorated_box.dart (hunk 2 above)
Comment (minhqdao) [reply lib/src/rules/use_decorated_box.dart]: It's been changed, and I also added myself to `AUTHORS` 😏

Comment (pq) [reply lib/src/rules/use_decorated_box.dart]: Thanks!

**Review by pq — APPROVED** [review]

**Pull request merged**